src/main/scala/com/databricks/labs/overwatch/validation/DeploymentValidation.scala

```diff
@@ -0,0 +1,582 @@
+    try {
+      dbutils.fs.mkdirs(s"""${config.etl_storage_prefix}/test_access""")
+      dbutils.fs.put(s"""${config.etl_storage_prefix}/test_access/testwrite""", "This is a file in cloud storage.")
+      dbutils.fs.head(s"""${config.etl_storage_prefix}/test_access/testwrite""")
+      dbutils.fs.rm(s"""${config.etl_storage_prefix}/test_access""", true)
+      DeploymentValidationReport(true,
+        getSimpleMsg("Storage_Access"),
+        testDetails,
+        Some("SUCCESS"),
+        Some(config.workspace_id)
+      )
+    }
+    catch {
+      case exception: Exception =>
+        val msg = s"""Unable to read/write/create file in the provided etl storage prefix"""
+        val fullMsg = PipelineFunctions.appendStackStrace(exception, msg)
+        logger.log(Level.ERROR, fullMsg)
+        if (fastFail) {
+          throw new BadConfigException(fullMsg)
+        }
+        DeploymentValidationReport(false,
+          getSimpleMsg("Storage_Access"),
+          testDetails,
+          Some(fullMsg),
+          Some(config.workspace_id)
+        )
+    }
+  }
+
+  private def validateMountCount(conf: MultiWorkspaceConfig): DeploymentValidationReport = {
+
+    val testDetails =
+      s"""WorkSpaceMountTest
+         |APIURL:${conf.api_url}
+         |DBPATWorkspaceScope:${conf.secret_scope}
+         |SecretKey_DBPAT:${conf.secret_key_dbpat}""".stripMargin
+    try {
+      val patToken = dbutils.secrets.get(scope = conf.secret_scope, key = conf.secret_key_dbpat)
+      val apiEnv = ApiEnv(false, conf.api_url, patToken, getClass.getPackage.getImplementationVersion)
+      val endPoint = "dbfs/search-mounts"
+      val mountCount = ApiCallV2(apiEnv, endPoint).execute().asDF().count()
+      if(mountCount<50)
+        {
+          DeploymentValidationReport(true,
+            getSimpleMsg("Validate_Mount"),
+            testDetails,
+            Some("SUCCESS"),
+            Some(conf.workspace_id)
+          )
+        }else{
+        DeploymentValidationReport(false,
+          getSimpleMsg("Validate_Mount"),
+          testDetails,
+          Some("Number of mounts found in workspace is more than 50"),
+          Some(conf.workspace_id)
+        )
+      }
+
+    } catch {
+      case exception: Exception =>
+        val msg =
+          s"""No Data retrieved
+             |WorkspaceId:${conf.workspace_id}
+             |APIURL:${conf.api_url}
+             | DBPATWorkspaceScope:${conf.secret_scope}
+             | SecretKey_DBPAT:${conf.secret_key_dbpat}""".stripMargin
+        val fullMsg = PipelineFunctions.appendStackStrace(exception, msg)
+        logger.log(Level.ERROR, fullMsg)
+        DeploymentValidationReport(false,
+          getSimpleMsg("Validate_Mount"),
+          testDetails,
+          Some(fullMsg),
+          Some(conf.workspace_id)
+        )
+
+    }
+
+  }
+
+
+
+  /**
+   * Performs clusters/list api call to check the access to the workspace
+   *
+   * @param row
```

Comment:
several comments are out of date. Please cleanup the comments across the PR.

Reply:
done